clang/test/SemaCXX/cheri/cheri-cap-ptr-casts.cpp

```diff
@@ -0,0 +1,28 @@
+// RUN: %cheri_cc1 %s -DTEST -verify=expected,hybrid
+// RUN: %cheri_purecap_cc1 %s -DTEST -verify=expected,purecap
+// RUN: %cheri_cc1 %s -fsyntax-only -ast-dump 2>/dev/null | FileCheck %s -check-prefixes AST,HYBRID-AST
+// RUN: %cheri_purecap_cc1 %s -fsyntax-only -ast-dump 2>/dev/null | FileCheck %s -check-prefixes AST,PURECAP-AST
```

Comment:
```suggestion
// RUN: %cheri_cc1 %s -verify=expected,hybrid -ast-dump | FileCheck %s -check-prefixes AST,HYBRID-AST
// RUN: %cheri_purecap_cc1 %s -verify=expected,purecap -ast-dump | FileCheck %s -check-prefixes AST,PURECAP-AST
```

As long as you keep the error at the end there should be no need for the `#ifdef TEST`. (tested locally)